src/Microsoft.ML.Data/Evaluators/BinaryClassifierEvaluator.cs

```diff
@@ -789,6 +789,34 @@ private void ComputePrCurves()
             }
         }
 
+        /// <summary>
+        /// Generates precision recall curve data points.
+        /// </summary>
+        /// <param name="data">The scored data.</param>
+        /// <param name="label">The name of the label column in <paramref name="data"/>.</param>
+        /// <param name="score">The name of the score column in <paramref name="data"/>.</param>
+        /// <param name="probability">The name of the probability column in <paramref name="data"/>, the calibrated version of <paramref name="score"/>.</param>
+        /// <param name="predictedLabel">The name of the predicted label column in <paramref name="data"/>.</param>
+        /// <returns>The precision recall curve data points.</returns>
+        public IDataView GetPrecisionRecallCurve(IDataView data, string label, string score, string probability, string predictedLabel)
```

Comment:
>predictedLabel [](start = 120, length = 14)

Does the `BinaryClassifierEvaluator` need that column to evaluate? (same question for the API below).

Reply:
Yes it is needed, for ex. for accuracy metrics.

---
In reply to: [267902785](https://github.com/dotnet/machinelearning/pull/3039#discussion_r267902785) [](ancestors = 267902785)

Reply:
Incidentally @yaeldekel is correct, the original command line didn't *need* it -- it would use it if required, but it would not insist on it. That we have it for the API is somewhat unfortunate, but not that big of a deal. In future we could in fact improve this story: we could have *this* overload, but also another overload that *doesn't* take it, and takes a threshold for it. Thereby capturing the existing functionality of this C# API, while retaining the "flexibility" we used to achieve in our command line.

That however is beyond the scope of @ganik's work here, and not something we should add now.

Reply:
thank you for explanation.

---
In reply to: [268327749](https://github.com/dotnet/machinelearning/pull/3039#discussion_r268327749) [](ancestors = 268327749)